rustls/src/client/tls12.rs

```diff
@@ -490,7 +490,14 @@ fn emit_certverify(
         Some(signer) => signer,
     };
 
-    let message = transcript.take_handshake_buf();
+    let message = match transcript.take_handshake_buf() {
+        Some(msg) => msg,
+        None => {
+            trace!("Not sending CertificateVerify, no transcript");
+            return Ok(());
+        }
```

Comment:
Can we match on both client_auth.signer.take() and transcript.take_handshake_buf() at once?

Reply:
(to achieve the aim of one piece of code that handles the not-sending CertificateVerify case)

Reply:
Yup, done.

Reply:
(This has two different takes on not-sending, let me know if you want to collapse those.)